app/decorators/current_user.rb

```diff
@@ -0,0 +1,46 @@
+class CurrentUser < Draper::Decorator
+  decorates :user
+  delegate_all
+
+  def session_url
+    if object
+      h.link_to("Logout",
+                h.logout_path,
+                method: :delete)
+    else
+      h.link_to("Login with GitHub",
+                "/auth/github",
+                method: :get)
+    end
+  end
+
+  def create_room_url
+    if object && !h.current_page?(h.new_room_path)
+      h.link_to("Create New Room",
+                h.new_room_path)
+    else
+      ""
+    end
+  end
+
+  def form_link(form)
+    if object
+      form.submit("Create room", class: "btn")
+    else
+      h.link_to("Login to create room",
+                "/auth/github",
+                class: "btn")
+    end
+  end
+
+  def close_topic(room)
+    if object == room.user
+      h.button_to("Close this Topic",
+                 h.room_path(room),
+                 method: :put,
+                 class: "btn col s12")
+    else
+      ""
+    end
+  end
+end
```

Comment:
You really feel like this file is a better approach than having an if/else in a view? It's much harder to understand and read through. Like what is `h`? Why are there `else` blocks that return empty strings? It's great to see a decorator used, but it just seems like a ton of overhead for something that doesn't need to be optimized.